### PR TITLE
Add examples (vanilla PyTorch, PyTorch Lightning, and PyTorch Lightning Distributed)

### PR DESCRIPTION
**Pull request by johnsutor**
This commit was proposed in #1451 and adds the appropriate examples and documentation for running those examples for the MMCR loss.

### CODE REVIEW

File: examples/pytorch/mmcr.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+from lightly.transforms.mmcr_transform import MMCRTransform
+from lightly.utils.scheduler import cosine_schedule
+
+
+class MMCR(nn.Module):
+    def __init__(self, backbone):
+        super().__init__()
+
+        self.backbone = backbone
+        self.projection_head = SimCLRProjectionHead(512, 512, 128)
```
Comment (guarin) [inline examples/pytorch/mmcr.py:23]: This reminds me that we should also add an `MMCRProjectionHead` with default dimensions (8192, 8192, 512). But this can be done in a different PR.

File: examples/pytorch_lightning/mmcr.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from torch import nn
+
+from lightly.loss import MMCRLoss
+from lightly.models.modules import SimCLRProjectionHead
+from lightly.models.utils import deactivate_requires_grad, update_momentum
+from lightly.transforms.mmcr_transform import MMCRTransform
+from lightly.utils.scheduler import cosine_schedule
+
+
+class BYOL(pl.LightningModule):
```
Comment (guarin) [inline examples/pytorch_lightning/mmcr.py:19]:
```suggestion
class MMCR(pl.LightningModule):
```

File: examples/pytorch_lightning/mmcr.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        z_m = torch.stack(z_m, dim=1)
+
+        loss = self.criterion(z_o, z_m)
+        return loss
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=0.06)
+
+
+model = BYOL()
```
Comment (guarin) [inline examples/pytorch_lightning/mmcr.py:63]:
```suggestion
model = MMCR()
```

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks a lot! I think you forgot to rename `BYOL` to `MMCR` in one place but otherwise everything looks good to me!

File: examples/pytorch/mmcr.py (hunk 1 above)
Comment (guarin) [reply examples/pytorch/mmcr.py:23]: Made an issue for it: https://github.com/lightly-ai/lightly/issues/1481

**Comment by guarin** [comment]
Thanks for the update! This is ready to merge. I made some small additions to make sure that the docs pages for the different examples and classes are actually shown once we publish the new docs.

**Review by guarin — APPROVED** [review]
